psi4/src/psi4/fisapt/fisapt.cc

```diff
@@ -118,13 +118,13 @@ void FISAPT::localize() {
     ranges.push_back(vectors_["eps_focc"]->dimpi()[0]);
     ranges.push_back(vectors_["eps_occ"]->dimpi()[0]);
 
-    std::shared_ptr<fisapt::IBOLocalizer2> local =
-        fisapt::IBOLocalizer2::build(primary_, reference_->get_basisset("MINAO"), matrices_["Cocc"], options_);
-    local->print_header();
-    std::map<std::string, std::shared_ptr<Matrix> > ret = local->localize(matrices_["Cocc"], Focc, ranges);
+    auto local = IBOLocalizer::build(primary_, reference_->get_basisset("MINAO"), matrices_["Cocc"], options_);
+    local->set_ranges(ranges);
+    local->localize();
+    local->fock_update(Focc);
```

Comment:
It appears that the `IBOLocalizer::build` call is always accompanied by exactly one call to `set_ranges` before localization and exactly one call to `fock_update` after localization. For this reason, I don't think `set_ranges` or `fock_update` should be user facing functions. They don't add any flexibility to the user, and they require the user to memorize a list of additional functional calls.

Perhaps `ranges` and `Focc` can be made arguments to `build`, `set_ranges` can be made to call `build`, and `localize` can be made to also call `fock_update`.

Reply:
Addressed